code/game/objects/structures/barsigns.dm

```diff
@@ -33,6 +33,9 @@
 	if(sign.rename_area && sign.name)
 		rename_area(src, sign.name)
 
+	if(sign.icon) // Fulpstation Barsigns edit
+		icon = sign.fulpbarsign || 'icons/obj/barsigns.dmi' // Fulpstation Barsigns edit
+
```

Comment:
```suggestion
	icon = sign.fulpbarsign ? 'fulp_modules/barsigns_port/barsigns.dmi' : 'icons/obj/barsigns.dmi' // Fulpstation Barsigns edit

```

Reply:
Why put "if(sign.icon)" at all? It's always true unless the datum is empty.
What I did here would set icon to Fulp file, if "fulpbarsign" isn't empty, otherwise it will use default /tg/ .dmi.

Reply:
Oh, and I'd make "fulpbarsign" a boolean if you ever consider using my proposal.